pmm-app/tests/pages/qanPage.js

```diff
@@ -129,43 +137,44 @@ module.exports = {
 
   getFilterLocator(filterValue) {
     const filterLocator =
-      "//section[@class='aside__filter-group']//span[contains(text(), '" +
-      filterValue +
-      "')]/../span[@class='checkbox-container__checkmark']";
+     "//section[@class='aside__filter-group']//span[contains(text(), '" +
+     filterValue +
+     "')]/../span[@class='checkbox-container__checkmark']";
     return filterLocator;
   },
 
-  applyFilter(filterValue) {
+  applyFilter(filterValue){
     const filterLocator = this.getFilterLocator(filterValue);
     I.waitForElement(filterLocator, 30);
     I.click(filterLocator);
     I.waitForVisible(this.fields.table, 30);
   },
 
+  async expandAllFilter() {
+    for (let i = 0; i < 4; i++) {
+      // eslint-disable-next-line max-len
+      let numOfElementsFilterCount = await I.grabNumberOfVisibleElements(
+        this.filterGroupCountSelector(this.filterGroups[i])
+      );
+      if (numOfElementsFilterCount === 1) {
+        // eslint-disable-next-line max-len
+        I.click(this.filterGroupCountSelector(this.filterGroups[i]));
+        // eslint-disable-next-line max-len
+        I.waitForVisible("//section[@class='aside__filter-group']//span[contains(text(), '" + this.filterGroups[i] + "')]/../button[contains(text(), 'Show top 5')]");
+      }
+    }
+  },
+
   async _getData(row, column) {
-    const percentage = await I.grabTextFrom(
-      "//table//tr[@ng-reflect-router-link='details/," +
-        (row - 1) +
-        "']//app-qan-table-cell[" +
-        column +
-        ']//div[1]//div[3]'
-    );
-    const value = await I.grabTextFrom(
-      "//table//tr[@ng-reflect-router-link='details/," +
-        (row - 1) +
-        "']//app-qan-table-cell[" +
-        column +
-        ']//div[1]//div[2]'
-    );
+    const percentage = await I.grabTextFrom("//table//tr[@ng-reflect-router-link='details/," + (row - 1) + "']//app-qan-table-cell[" + column +"]//div[1]//div[3]");
```

Comment:
You could you much simpler approach here by using template string and locators functions (function names just for example ):

Create locator functions in page object:

percentageLocator(row, column) {
return `//table//tr[@ng-reflect-router-link='details/,${row - 1}']//app-qan-table-cell[${column}]//div[1]//div[3]`
}
valueLocator(row, column) {
return `//table//tr[@ng-reflect-router-link='details/,${row - 1}']//app-qan-table-cell[${column}]//div[1]//div[2]`
}

And use it:
`
const percentage = await I.grabTextFrom(percentageLocator(row, column));
const value = await I.grabTextFrom(valueLocator(row, column));
`

Reply:
This changes may be applied to all generated selectors

Reply:
I am not sure, but I didn't really introduce this variable, it was preexisting, all I did was to make it more better to read and not break it in 4 different lines

Reply:
when we add new QAN, this will be completely changed, so do you want me to make the change for this now? I already used the concept, I use Locators Function to expand all filters now.

Reply:
Agree with you, It may wait till new qan release, let's merge this one to not spend too much time on it